### DISCUSSION TITLE
Domain

### DISCUSSION THREAD

**Discussion by lroal** [cloudflare/workerd#3125] [Node.js APIs]
I was under the impression that Domain was supported when running with latest compatability flags.
But I got 
```
 Domain.run is not implemented yet!
```

**Comment by lroal** [comment]
Async local storage / async hooks is not an option - as it is unstable.

**Reply by anonrig** [reply]
What specific functionality do you use from node:domain?

**Reply by lroal** [reply]
Domain.run and process.domain.
I am using it for "thread singletons" / async context.
I am library author of https://github.com/alfateam/orange-orm and currenly creating an adapter for Cloudflare D1.
